### PR TITLE
Backport v1 codetree from v2.3.0 as v1.2.3

### PR DESCRIPTION
**Pull request by wallyqs**
This includes the v1 version with the vulnerability fixes from v2.3.0 as a version named "v1.2.3" which is one patch version on top of the last release from `nats-io/jwt@v1.2.2`. Main goal from this is to prevent `nsc` bringing the unused bad layer via the go modules pseudo version to instead use a tagged version: https://github.com/nats-io/nsc/blob/main/go.mod#L12

Fixes https://github.com/nats-io/nsc/issues/535

Signed-off-by: Waldemar Quevedo <wally@nats.io>

### CODE REVIEW

**Comment by wallyqs** [comment]
Once merged, I will release `v1.2.3` and then have that as the version that `nsc` v2 imports.

File: imports_test.go [hunk 1]
```diff
@@ -157,7 +135,7 @@ func TestInvalidImportTokenValuesValidation(t *testing.T) {
 	activation.Expires = time.Now().Add(time.Duration(time.Hour)).UTC().Unix()
 
 	activation.ImportSubject = "test"
-	activation.ImportType = Stream
+	activation.ImportType = Service
```
Comment (aricart) [inline imports_test.go:138]: why is this test changing from Stream to Service?

**Review by aricart — COMMENTED** [review]
Kind of worried about some of the changes to the v1 source. But at the same time, one question is whether we should support them in a context other than upgrading the JWT...

**Comment by wallyqs** [comment]
We need a release on top o v1.2.2 with the security fixes so that `nsc` binary does not get flagged by scanners, or remove the dependency from `nsc` here: https://github.com/nats-io/nsc/blob/main/go.mod#L12

**Comment by aricart** [comment]
The code trees are not 1-1, there were some inversions etc - so porting over changes from one tree to the other may not do the right thing.

**Comment by wallyqs** [comment]
ok we just need a tagged v1.2.X release that does not has the issue then make `nsc` depend on that one

**Comment by wallyqs** [comment]
Closing since this is a better workaround: https://github.com/nats-io/nsc/pull/564#pullrequestreview-1320689469

**Pull request closed without merging**